lib/AST/ASTPrinter.cpp

```diff
@@ -907,14 +907,11 @@ class PrintAST : public ASTVisitor<PrintAST> {
         DC = DC->getParent()->getInnermostTypeContext();
```

Comment:
What do you mean by two levels of nested type? If I'm getting the member type of A.B.C, I have to pass in A.B as the DC, which is what you get if you call getDeclContext() on the NominalTypeDecl for A.B.C.

The logic for 'checking' the dc is wrong, period. We need to find a better fix.